### DISCUSSION TITLE
Using LaravelData from within another Package

### DISCUSSION THREAD

**Discussion by robmellett** [spatie/laravel-data#498] [Q&A] [answered]
Hello!

Thanks for your awesome work with this.

I was just wondering if there was a recommended approach to using this package from within another package (say the spatie/laravel-package-skeleton).

After installing the package `composer require spatie/laravel-data` and copying over the configuration file.

An exception is thrown when using a when creating a new Data object.

**Example**
```
new SongData('Never gonna give you up', 'Rick Astley');
```

**Will produce.**
```
  Unresolvable dependency resolving [Parameter #0 [ <required> array $config ]] in class Spatie\LaravelData\Support\DataConfig
```

After looking at the `https://github.com/spatie/laravel-data/blob/main/src/LaravelDataServiceProvider.php` class, I can see that adding the following two methods `packageRegistered` and `packageBooted` to our package skeleton ServiceProvider will work - but I was just wondering if there was a smoother way of course.

Any ideas or suggestions would be much appreciated.

<3

**Comment by rubenvanassche** [comment, marked as answer]
Do you get these errors within your package tests? Or in the application where both packages are included? In the first case, within the orchestral test suite you can boot up additional service providers. Adding LaravelDataServiceProvider here should normally fix the problem.

**Reply by robmellett** [reply]
Thank you so much @rubenvanassche <3

Registering the `laravel-data` within my orchestral test suite was exactly what I needed.

I owe you a beer :)
